tests/results/test_disability_observer.py

```diff
@@ -230,3 +235,125 @@ def test_disability_accumulation(
             ].values
             assert len(actual_ylds) == 1
             assert np.isclose(expected_ylds, actual_ylds[0], rtol=0.0000001)
+
+
+@pytest.mark.parametrize("exclusions", [[], ["flu"], ["all_causes"], ["flu", "all_causes"]])
+def test_set_causes_of_disability(exclusions, mocker):
+    observer = DisabilityObserver_()
+
+    builder = mocker.Mock()
+    mocker.patch("vivarium.component.Component.build_all_lookup_tables")
+    builder.configuration.time.step_size = 28
+    builder.configuration.stratification.excluded_categories = LayeredConfigTree(
+        {"disability": exclusions}
+    )
+
+    # Set up fake calls for cause-specific register_observation args
+    flu = DiseaseState("flu")
+    builder.components.get_components_by_type = lambda n: [flu]
+
+    observer.setup_component(builder)
+    assert {cause.state_id for cause in observer.causes_of_disability} == {
+        "flu",
+        "all_causes",
+    } - set(exclusions)
+
+
+def test_set_causes_of_disability_raises(mocker):
+    observer = DisabilityObserver_()
+
+    builder = mocker.Mock()
+    mocker.patch("vivarium.component.Component.build_all_lookup_tables")
+    builder.configuration.time.step_size = 28
+    builder.configuration.stratification.excluded_categories = LayeredConfigTree(
+        {"disability": ["arthritis"]}  # not an instantiated disease
+    )
+
+    # Set up fake calls for cause-specific register_observation args (but NOT 'arthritis')
+    flu = DiseaseState("flu")
+    builder.components.get_components_by_type = lambda n: [flu]
+
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "Excluded 'disability' causes {'arthritis'} not found in expected categories categories: ['flu', 'all_causes']"
+        ),
+    ):
+        observer.setup_component(builder)
+
+
+# FIXME: error when excluding all but one disability
+@pytest.mark.parametrize("exclusions", [[], ["all_causes"], ["all_causes", "sick_cause_0"]])
+def test_category_exclusions(
+    base_config,
+    base_plugins,
+    exclusions,
+):
+    """Integration test for the disability observer and the Results Management system."""
+
+    if len(exclusions) == 2:
+        pytest.skip(reason="FIXME: error when excluding all but one disability")
```

Comment:
Not sure how high of a priority this is. Seems unlikely that we'd every exclude evertyhing but a single disability? Will make a separate ticket though b/c this PR has been open long enough.

Reply:
rajan confirmed it does not have to be part of this PR